test/test_api/test_base_api.py

```diff
@@ -20,6 +20,7 @@
 # ====
 @pytest.mark.parametrize("fit_dictionary_tabular", ['classification_categorical_only'], indirect=True)
 def test_nonsupported_arguments(fit_dictionary_tabular):
+    BaseTask.__abstractmethods__ = set()
```

Comment:
Interesting way to do this, I normally make a `MockBaseTask` class in some `mocks.py`, didn't know you could do this

Reply:
yeah I read that having `abstractmethods` is what causes an error while instantiating an abstract class. I thought for testing its fine to have this workaround.